### PR TITLE
Upgrade to Netcoreapp3.0 preview9

### PR DESCRIPTION
**Pull request by bergmeister**
# PR Summary

Update to .Net Core 3.0-preview9, including out of band NuGet packages
cc @SteveL-MSFT 

## PR Context

https://devblogs.microsoft.com/dotnet/announcing-net-core-3-0-preview-9/

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj [hunk 1]
```diff
@@ -16,11 +16,11 @@
...
+    <PackageReference Include="System.Data.SqlClient" Version="4.7.0-preview9.19421.4" />
     <!-- the following package(s) are from https://github.com/dotnet/corefx -->
-    <PackageReference Include="System.IO.Packaging" Version="4.6.0-preview8.19405.3" />
-    <PackageReference Include="System.Net.Http.WinHttpHandler" Version="4.6.0-preview8.19405.3" />
-    <PackageReference Include="System.Text.Encodings.Web" Version="4.6.0-preview8.19405.3" />
+    <PackageReference Include="System.IO.Packaging" Version="4.6.0-preview9.19421.4" />
+    <PackageReference Include="System.Net.Http.WinHttpHandler" Version="4.6.0-preview9.19421.4" />
+    <PackageReference Include="System.Text.Encodings.Web" Version="4.6.0-preview9.19421.4" />
     <!-- the following package(s) are from https://github.com/dotnet/wcf -->
     <PackageReference Include="System.ServiceModel.Duplex" Version="4.6.0-preview8.final" />
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj]: Why aren't we updating this and some packages below?

File: test/tools/WebListener/WebListener.csproj [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview6.19307.2" />
```
Comment (iSazonov) [inline test/tools/WebListener/WebListener.csproj]: Quite old too.

Comment (bergmeister) [reply test/tools/WebListener/WebListener.csproj]: Thanks, I've updated it now

File: src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj (hunk 1 above)
Comment (bergmeister) [reply src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj]: Thanks, I've updated those `*.final` packages now as well

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj [hunk 3]
```diff
@@ -56,7 +56,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.2.1" />
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: Not for the PR - I wonder that we don't update this to 3.2.0 https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/3.2.0

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj (hunk 3 above)
Comment (bergmeister) [reply src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: Why? That would be a downgrade.

**Review by daxian-dbw — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj (hunk 3 above)
Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: I think he meant: https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/3.3.1-beta3-final
image

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan 	Please remind me in 1 hour

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj (hunk 3 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: Yes, sorry.

**Pull request merged**